src/libraries/System.Net.Requests/src/System/Net/HttpWebRequest.cs

```diff
@@ -1202,6 +1205,100 @@ private async Task<WebResponse> SendRequest(bool async)
             }
         }
 
+        private void AddCacheControlHeaders(HttpRequestMessage request)
+        {
+            RequestCachePolicy? policy = GetApplicableCachePolicy();
+
+            if (policy != null)
+            {
+                if (request.Headers.CacheControl == null)
+                {
+                    request.Headers.CacheControl = new CacheControlHeaderValue();
+                }
+
+                if (policy is HttpRequestCachePolicy httpRequestCachePolicy)
+                {
+                    switch (httpRequestCachePolicy.Level)
+                    {
+                        case HttpRequestCacheLevel.NoCacheNoStore:
+                            request.Headers.CacheControl.NoCache = true;
+                            request.Headers.CacheControl.NoStore = true;
+                            request.Headers.Pragma.Add(new NameValueHeaderValue("no-cache"));
+                            break;
+                        case HttpRequestCacheLevel.Reload:
+                            request.Headers.CacheControl.NoCache = true;
+                            request.Headers.Pragma.Add(new NameValueHeaderValue("no-cache"));
+                            break;
+                        case HttpRequestCacheLevel.CacheOnly:
```

Comment:
Same here.

(note: the `CacheOrNextCacheOnly` behavior is correct, it's just `CacheOnly` that should throw.)